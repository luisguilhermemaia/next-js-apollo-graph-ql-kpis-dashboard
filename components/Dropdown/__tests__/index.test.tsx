import '@testing-library/jest-dom/extend-expect';
import { RenderResult, render } from '@testing-library/react';
import React from 'react';
import { Dropdown } from '../index';

describe('Dropdown', () => {
  let renderResult: RenderResult;
  const items = [
    {
      value: 'THIS_WEEK',
      label: 'This Week',
      payload: { timeUnit: 'WEEK', timeUnitCount: 0 },
    },
    {
      value: 'LAST_WEEK',
      label: 'Last Week',
      payload: { timeUnit: 'WEEK', timeUnitCount: 1 },
    },
  ];
  const doRender = () => {
    renderResult = render(
      <Dropdown
        setState={jest.fn()}
        items={items}
        value={items[0]}
        label="Filter By"
      />
    );
  };

  it('renders correctly', () => {
    doRender();

    const { container } = renderResult;

    expect(container).toMatchSnapshot();
  });
});
