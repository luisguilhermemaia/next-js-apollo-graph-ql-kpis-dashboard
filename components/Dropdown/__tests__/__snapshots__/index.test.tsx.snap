// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly 1`] = `
<div>
  <div
    class="Dropdown__DropdownWrapper-y8i6hw-1 YfFJY"
  >
    <span
      class="Dropdown__StyledLabel-y8i6hw-0 sZwBi"
    >
      Filter By
    </span>
    <input
      class="Dropdown__StyledButton-y8i6hw-2 koLxyO"
      type="button"
      value="This Week"
    />
    <ul
      class="Dropdown__StyledButtonUl-y8i6hw-3 dwFUEW"
    >
      <li
        class="Dropdown__StyledButtonLi-y8i6hw-4 bcOFik"
      >
        This Week
      </li>
      <li
        class="Dropdown__StyledButtonLi-y8i6hw-4 bcOFik"
      >
        Last Week
      </li>
    </ul>
  </div>
</div>
`;
